### ISSUE TITLE
Completely disable TLS1.0 from NGINX

### ISSUE THREAD

**Issue by rosch100** [nginx-proxy/nginx-proxy#1177]
When testing my web-servers using ssllabs' server test, I always got a result, which stated that TLS 1.0 was enabled. This was confirmed by 
`openssl s_client -connect my.host:443 -tls1`
which could connect successfully.

This behavior was independent from my SSL_POLICY settings.

After some trial-and-error I came up with a solution following [https://serverfault.com/questions/704376/disable-tls-1-0-in-nginx](url)

After adding `default_server` and `ssl_protocols` options in the `server_name _` section in nginx.tmpl:
```
{{ if (and (exists "/etc/nginx/certs/default.crt") (exists "/etc/nginx/certs/default.key")) }}
server {
        server_name _; # This is just an invalid value which will never trigger on a real hostname.
        listen 443 ssl http2 default_server;
        {{ if $enable_ipv6 }}
        listen [::]:443 ssl http2 default_server;
        {{ end }}
        ssl_protocols TLSv1.2 TLSv1.3;
```
I was not able to connect using TLS 1.1 or TLS 1.0 anymore.

**Comment by jgimenez** [comment]
In case it's useful for someone, I figured out how to configure it without changing the template. As explained above, the problem is that the configuration taken is the one from the default virtualhost. If no virtualhost is defined as default, then the configuration specified is just ignored. This works for me:

```
services:
  ssl:
    image: jwilder/nginx-proxy
    ports:
      - 80:80
      - 443:443
    environment:
      - DEFAULT_HOST=my.site.com
  mysite:
    image: xxx
    environment:
      - SSL_POLICY=Mozilla-Modern
      - VIRTUAL_HOST=my.site.com
```

**Comment by Nottt** [comment]
@jgimenez what if you do don't want to have a "default_host"? I serve multiple websites on that server...  and my understanding is that if for some reason the website is down, nginx will serve the default host which could lead to a lot of strange issues...

Example: User visits site A which is down, nginx servers website B (default host) causing possibly a SSL issue or just confusing the user...

**Comment by buchdag** [comment]
This should have been fixed by #2186 and #2338
